### PR TITLE
docs: Infrastructure updates, add howto for making a release

### PR DESCRIPTION
**Pull request by paulgessinger**
This PR:

1. Switches to MyST from recommonmark + m2r. The latter are deprecated and MyST should be an improvement all around (directives possible in markdown now)
2. Adds a navigation point for the existing top level how-to pages
3. Adds a how-to for making a release

### CODE REVIEW

File: docs/howto/release.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+```{note}
+This assumes you have a local clone of the ACTS repository at hand.
+You'll also need write permissions on the upstream ACTS repository to be able to push to the `releases` branch.
+```
+
+
+```console
+$ git checkout releases
+$ git fetch upstream # make sure you're up to date
+$ git merge --no-ff origin/main
```
Comment (andiwand) [inline docs/howto/release.md:18]: should this be `git merge --no-ff upstream/main` ?

File: docs/howto/release.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+29f3649 2022-06-07 14:22 +0200 Noemi Calace               │ o fix: Fixing typo in name of variable and associated methods (#1277)
+239ece3 2022-06-03 17:39 +0200 robertlangenberg           │ o  bump macos dependencies (#1279)
+f77f377 2022-05-30 15:46 +0200 Noemi Calace               │ o feat: Adding features for sorting space points in r in (phi, z) grid (#1267)
+a7ee09d 2022-05-25 11:17 +0200 Luis Falda Coelho          │ o fix: Bug in xyz strip coordinates transformation  (#1265)
+82f42a2 2022-05-25 09:02 +0000 github-actions[bot]        o │ <v19.1.0> Bump to version v19.1.0
+7c59e12 2022-05-25 10:26 +0200 Paul Gessinger             M─│─┐ Merge remote-tracking branch 'origin/main' into releases
+4ceddf3 2022-05-25 10:26 +0200 Luis Falda Coelho          │ o─┘ feat: ITk seedFilter integration and seed quality confirmation  (#1201)
+```
+
+You can now push the updated `releases` branch to the remote `releases` branch using `git push -u origin releases`.
```
[the remark this answers is not in the document]
Comment (paulgessinger) [reply docs/howto/release.md:48]: Sure. I name my upstreams `origin`, that's why I have this here.

**Review by AJPfleger — APPROVED** [review]
lgtm

**Comment by paulgessinger** [comment]
@AJPfleger or @andiwand can you reapprove?

**Review by andiwand — APPROVED** [review]

**Pull request merged**